### PR TITLE
Watchlater page

### PR DESCRIPTION
**Pull request by agrawalshanu708**
create a watchlater page,
User can add or remove the items in page

### CODE REVIEW

File: src/components/navbar/navbar.jsx [hunk 1]
```diff
@@ -1,19 +1,20 @@
...
 const Navbar = () => {
     const{likeState} = useLike()
+    const{watchLaterState} = useWatchLater()
   return (
     <>
     <div className="navbar">
-<div className="navbar__logo">
+<Link to = "/"> <div className="navbar__logo">
     <div className="navbar__media">
         <img src={logo} alt="" />
```
Comment (Chiragmodi01) [inline src/components/navbar/navbar.jsx:14]: don't leave alt empty

File: src/components/videostore/videocard.jsx [hunk 2]
```diff
@@ -2,10 +2,12 @@ import React from 'react'
 import {AiOutlineHeart} from "react-icons/ai"
 import {MdOutlinePlaylistPlay} from "react-icons/md"
 import {MdOutlineWatchLater} from"react-icons/md"
```
Comment (Chiragmodi01) [inline src/components/videostore/videocard.jsx:4]: Import icons from same library together, {MdOutlinePlaylistPlay, MdOutlineWatchLater}

File: src/components/videostore/videocard.jsx [hunk 3]
```diff
@@ -25,7 +27,12 @@ const {likeState,likeDispatch} = useLike()
         payload: {_id,title,description,charactor}
```
Comment (Chiragmodi01) [inline src/components/videostore/videocard.jsx:27]: You're anyway using all these properties below while dispatching, so no need to destructure here.

File: src/index.js [hunk 4]
```diff
@@ -4,7 +4,7 @@ import "./index.css";
 import App from "./App";
 import { makeServer } from "./server";
 import { BrowserRouter } from "react-router-dom";
-import { LikeProvider } from "./context/likecontext";
+import { LikeProvider,WatchLaterProvider } from "./context/index";
 
 
 // Call make Server
```
Comment (Chiragmodi01) [inline src/index.js:10]: No relative code for this, so remove this comment.

File: src/pages/watchlaterpage/watchlater.jsx [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import React from 'react'
+import "./watchlater.css"
+import {useWatchLater} from "./../../context/index"
+import {WatchLaterCard} from "./watchlatercard"
+
+const WatchLaterPage = () => {
+    const{watchLaterState} = useWatchLater()
+  return (
+    <>
```
Comment (Chiragmodi01) [inline src/pages/watchlaterpage/watchlater.jsx:9]: Fragment is not necessary here.

File: src/pages/watchlaterpage/watchlatercard.jsx [hunk 6]
```diff
@@ -0,0 +1,34 @@
+import React from 'react'
+import {AiOutlineHeart} from "react-icons/ai"
+import {MdOutlinePlaylistPlay} from "react-icons/md"
+import {MdOutlineWatchLater} from"react-icons/md"
```
Comment (Chiragmodi01) [inline src/pages/watchlaterpage/watchlatercard.jsx:4]: Again, import both of them together.

**Review by Chiragmodi01 — COMMENTED** [review]
- Check if video exist in watchlater, if true then don't add.
- Keep some margin for man container in the UI.
- I'd prefer putting those nav-link's badges in circle div rather than letting it float in the air.

Overall code structure looks good, just added few sanity comment. You might want to refractor UI, fix spacing.

**Review by therajatg — APPROVED** [review]

File: src/reducer/watchlaterreducer.jsx [hunk 7]
```diff
@@ -0,0 +1,12 @@
+const watchLaterReducer = (watchLaterState,watchLaterAction) => {
+    switch (watchLaterAction.type) {
+        case "WATCH_LATER":
+   return {...watchLaterState, 
+  watchLaterItems: [...watchLaterState.watchLaterItems, watchLaterAction.payload]
+}   
+
+default:
+            return watchLaterState
```
Comment (Sharmaryan) [inline src/reducer/watchlaterreducer.jsx:9]: remove space

**Review by Sharmaryan — APPROVED** [review]
code looks clean and chirag already suggested some changes, you can make them and UI looks nice

**Pull request merged**